ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -170,7 +165,10 @@ object Trees {
     val tpe = NoType // cannot be in expression position
   }
 
-  case class Try(block: Tree, errVar: Ident, handler: Tree, finalizer: Tree)(
+  case class TryCatch(block: Tree, errVar: Ident, handler: Tree)(
+      val tpe: Type)(implicit val pos: Position) extends Tree
+
+  case class TryFinally(block: Tree, finalizer: Tree)(
       val tpe: Type)(implicit val pos: Position) extends Tree
```

Comment:
A `TryFinally` does not need an explicit `val tpe`. It's `tpe` is always `block.tpe`.